django/curator/models.py

```diff
@@ -300,3 +309,45 @@ def migrate(self, new_names, old_name):
             for model in through_models:
                 self.copy_through_model_refs(model, new_tags=new_tags, old_tag=old_tag)
             old_tag.delete()
+
+
+class UserSpamStatus(models.Model):
+    member_profile = models.OneToOneField(
+        MemberProfile, on_delete=models.CASCADE, primary_key=True
+    )
+    # FIXME: add help_text
+    # None = not processed yet
+    # True = bio_classifier considered this user to be spam
+    # False = bio_classifier did not consider this user to be spam
+    labelled_by_text_classifier = models.BooleanField(default=None, null=True)
+    text_classifier_confidence = models.FloatField(default=0)
+
+    # similar to bio_classifier
+    labelled_by_user_classifier = models.BooleanField(default=None, null=True)
+    user_classifier_confidence = models.FloatField(default=0)
+
+    labelled_by_curator = models.BooleanField(default=None, null=True)
+    last_updated = models.DateField(auto_now=True)
+    is_training_data = models.BooleanField(default=False)
+
+    @staticmethod
+    def get_recommendations_sorted_by_confidence():
+        return UserSpamStatus.objects.all().order_by("text_classifier_confidence")
+
+    def __str__(self):
+        return "user={}, labelled_by_text_classifier={}, text_classifier_confidence={}, labelled_by_user_classifier={}, user_classifier_confidence={}, labelled_by_curator={}, last_updated={}, is_training_data={}".format(
+            str(self.member_profile),
+            str(self.labelled_by_text_classifier),
+            str(self.text_classifier_confidence),
+            str(self.labelled_by_user_classifier),
+            str(self.user_classifier_confidence),
+            str(self.labelled_by_curator),
+            str(self.last_updated),
+            str(self.is_training_data),
+        )
+
+
+# Create a new UserSpamStatus whenever a new MemberProfile is created
+@receiver(post_save, sender=MemberProfile)
+def sync_member_profile_spam(sender, instance, **kwargs):
```

Comment:
I believe we need to check the `created` argument https://docs.djangoproject.com/en/4.2/ref/signals/#post-save otherwise this will try and fail to create a second object when a memberprofile gets updated